pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -935,11 +935,11 @@ self: super: builtins.intersectAttrs super {
 
   rel8 = addTestToolDepend super.rel8 pkgs.postgresql;
 
-  cachix = generateOptparseApplicativeCompletion "cachix" (super.cachix.override { nix = pkgs.nix_2_3; });
+  cachix = addBuildDepends (generateOptparseApplicativeCompletion "cachix" (super.cachix.override { nix = pkgs.nix_2_4; })) [ pkgs.nlohmann_json ];
```

Comment:
Maybe `nix_2_4` should propagate `nhlohmann_json`?

Reply:
I'll give it a shot.

Reply:
I forgot that my earlier solution (in the upstream repo) was a workaround. `nlohmann_json` has been vendored. I propose to add the headers to the output here https://github.com/NixOS/nix/pull/5536